### ISSUE TITLE
Floppy disk "bad sector" emulation

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#444]
It just occurred to me while watching "Ancient DOS Games" that it might help old copy-protected titles if DOSBox-X could accept a list of bad sectors to emulate alongside a floppy disk image.

The list would be a simple list of sectors (C/H/S) that should return an error on read instead of returning the data.

It was common for titles in the 80s to have deliberate bad sectors to detect copies.

**Comment by sikthehedgehog** [comment]
I was under the impression that the way those worked was by just marking the sectors as bad in the FAT (rather than actually being bad sectors), so using a stock FAT driver would report them as unusable but parsing the filesystem on your own could let you by-pass that (effectively making them uncopyable through naive means).

Actually making sectors bad probably would have been a pain to pull off at the factory. Though I guess it's possible somebody tried malformed sectors (the same way some floppies have non-standard sector sizes to squeeze in some more KB).

**Comment by joncampbell123** [comment]
@sikthehedgehog No. If it was just a "bad sector" in the FAT, a 1:1 copy of the disk would preserve that, FAT table included.

**Comment by joncampbell123** [comment]
http://diskpreservation.com/dp.php?pg=protection_atari

http://www.sierrahelp.com/AGI/CopyProtection/index.html

http://www.os2museum.com/wp/lotus-1-2-3-r3-copy-protection/

**Comment by rderooy** [comment]
Would it not be better to support an existing disk format that supports copy protected disks instead?

Looking around there is Teledisk (TD0), ImageDisk (IMD), Snatchit (CP2) which are able to support some, but not all copy protected disk formats apparently.

And then there is TransCopy (IMG), which supports most copy protections it seems, and lastly there is Kryoflux but that one creates rather large (and numerous) files.

**Comment by sofakng** [comment]
I'd also love to see support for Kryoflux or HFE(v3) image formats.  These formats allow for all (?) copy protection methods to be supported.

I don't like using cracks/patches if I don't need to and this support would be fantastic.

**Comment by joncampbell123** [comment]
I could do Kryoflux (read only at least), I have one of those and I've been imaging most of my floppies with it lately. I even wrote my own code to parse the raw flux format their tool puts out: https://github.com/joncampbell123/floppytools

**Comment by sofakng** [comment]
> I could do Kryoflux (read only at least), I have one of those and I've been imaging most of my floppies with it lately. I even wrote my own code to parse the raw flux format their tool puts out: https://github.com/joncampbell123/floppytools

That would be fantastic!  read-only is probably all that anybody really needs.

If you need any help testing I also have several images I could use for testing.

Thanks again very much!
